packages/platform-server/test/incremental_hydration_spec.ts

```diff
@@ -1514,5 +1520,233 @@ describe('platform-server partial hydration integration', () => {
       expect(appHostNode.outerHTML).toContain('<span>Client!</span>');
       expect(appHostNode.outerHTML).not.toContain('>Server!</span>');
     }, 100_000);
+
+    it('should clear registry of blocks as they are hydrated', async () => {
+      @Component({
+        standalone: true,
+        selector: 'app',
+        template: `
+          <main (click)="fnA()">
+            @defer (on viewport; hydrate on interaction) {
+              <div id="main" (click)="fnA()">
+                Main defer block rendered!
+                @defer (on viewport; hydrate on interaction) {
+                  <p id="nested">Nested defer block</p>
+                } @placeholder {
+                  <span>Inner block placeholder</span>
+                }
+              </div>
+            } @placeholder {
+              <span>Outer block placeholder</span>
+            }
+          </main>
+        `,
+      })
+      class SimpleComponent {
+        fnA() {}
+
+        registry: DehydratedBlockRegistry = inject(DEHYDRATED_BLOCK_REGISTRY);
+        jsActionMap: Map<string, Set<Element>> = inject(JSACTION_BLOCK_ELEMENT_MAP);
+        contract: EventContractDetails = inject(JSACTION_EVENT_CONTRACT);
```

Comment:
nit: those types are likely inferred by TS:
```suggestion
        registry = inject(DEHYDRATED_BLOCK_REGISTRY);
        jsActionMap = inject(JSACTION_BLOCK_ELEMENT_MAP);
        contract = inject(JSACTION_EVENT_CONTRACT);
```